cpp/src/c_api/renumber_arbitrary_edgelist.cu

```diff
@@ -0,0 +1,215 @@
+ */
+
+#include "c_api/resource_handle.hpp"
+#include "c_api/utils.hpp"
+#include "cugraph/graph.hpp"
+#include "cugraph/utilities/error.hpp"
+#include "cugraph_c/error.h"
+#include "thrust/binary_search.h"
+#include "thrust/iterator/counting_iterator.h"
+#include "thrust/iterator/zip_iterator.h"
+
+#include <cugraph_c/graph_functions.h>
+
+namespace {
+
+template <typename vertex_t>
+cugraph_error_code_t renumber_arbitrary_edgelist(
+  raft::handle_t const& handle,
+  cugraph::c_api::cugraph_type_erased_host_array_view_t const* renumber_map,
+  cugraph::c_api::cugraph_type_erased_device_array_view_t* srcs,
+  cugraph::c_api::cugraph_type_erased_device_array_view_t* dsts)
+{
+  // Create a sorted representation of each vertex id and where it exists in the input array
+  rmm::device_uvector<vertex_t> srcs_v(srcs->size_, handle.get_stream());
+  rmm::device_uvector<vertex_t> dsts_v(dsts->size_, handle.get_stream());
+  rmm::device_uvector<size_t> srcs_pos(srcs->size_, handle.get_stream());
+  rmm::device_uvector<size_t> dsts_pos(dsts->size_, handle.get_stream());
+
+  thrust::copy_n(handle.get_thrust_policy(), srcs->as_type<vertex_t>(), srcs->size_, srcs_v.data());
+  thrust::copy_n(handle.get_thrust_policy(), dsts->as_type<vertex_t>(), dsts->size_, dsts_v.data());
+  thrust::sequence(handle.get_thrust_policy(), srcs_pos.begin(), srcs_pos.end(), size_t{0});
+  thrust::sequence(handle.get_thrust_policy(), dsts_pos.begin(), dsts_pos.end(), size_t{0});
+
+  thrust::sort(handle.get_thrust_policy(),
+               thrust::make_zip_iterator(srcs_v.begin(), srcs_pos.begin()),
+               thrust::make_zip_iterator(srcs_v.end(), srcs_pos.end()));
+
+  thrust::sort(handle.get_thrust_policy(),
+               thrust::make_zip_iterator(dsts_v.begin(), dsts_pos.begin()),
+               thrust::make_zip_iterator(dsts_v.end(), dsts_pos.end()));
+
+  // Read chunk of renumber_map in a loop, updating base offset to compute vertex id
+  // FIXME: Compute this as a function of free memory?  Or some value that keeps a
+  // particular GPU saturated?
+  size_t chunk_size = size_t{1} << 20;
+
+  rmm::device_uvector<vertex_t> renumber_chunk(chunk_size, handle.get_stream());
+
+  for (size_t chunk_base_offset = 0; chunk_base_offset < renumber_map->size_;
+       chunk_base_offset += chunk_size) {
+    size_t size = std::min(chunk_size, renumber_map->size_ - chunk_base_offset);
+    if (size < chunk_size) renumber_chunk.resize(size, handle.get_stream());
+
+    raft::update_device(renumber_chunk.data(),
+                        renumber_map->as_type<vertex_t>() + chunk_base_offset,
+                        size,
+                        handle.get_stream());
+
+    rmm::device_uvector<vertex_t> renumbered_values(srcs_v.size(), handle.get_stream());
+
+    thrust::fill(handle.get_thrust_policy(),
+                 renumbered_values.begin(),
+                 renumbered_values.end(),
+                 cugraph::invalid_vertex_id<vertex_t>::value);
```

Comment:
Where are you using `renumbered_values`?

Reply:
Deleted.